snyk_sync/utils.py

```diff
@@ -298,18 +302,26 @@ def ensure_dir(directory: Path) -> bool:
 
 def load_watchlist(cache_dir: Path) -> SnykWatchList:
     tmp_watchlist = SnykWatchList()
+    cache_data_errors = []
 
     if path.exists(f"{cache_dir}/data.json"):
```

Comment:
Do you need "if path.exists(f"{cache_dir}/data.json"):" as you jopen and put that inside a try block, if the file doesnt exist will it get trap in the exception handling.

Reply:
yes, agreed.  from the looks of it, if the file doesn't exist it will just return an empty watchlist so maybe handling the file not found exception, writing to the console "could not load watchlist" and returning the function with the empty watchlist still, so the functionality is not changed.  I can make that change

Reply:
i dont think the `except KeyError as e:` line even does anything in practice the way its currently written

Reply:
i refactored this and ran basic tests, how does it look?